evaltools/data/estimatecvap.py

```diff
@@ -0,0 +1,258 @@
+    users specify a triple \((X, Y, Z)\) where \(X\) is the old CVAP column for
+    that group, \(Y\) is the old VAP column for that group, and Z is the new VAP
+    column for that group (\(Z\) must be a column in our base geodataframe). Then,
+    the estimated new CVAP for that group will be constructed by multiplying
+    \((X / Y) \cdot Z\) for each new geometry.
+
+    <div style="text-align: center;">
+        </br>
+        <img width="75%" src="../images/cvap-estimation.png"/>
+    </div>
+
+    Args:
+        base (GeoDataFrame): A `GeoDataFrame` with the appropriate columns for
+            estimating CVAP.
+        state (State): The `us.State` object for which CVAP data is retrieved.
+        groups (list): `(X, Y, Z)` triples for each desired CVAP group to be
+            estimated, where each of the parameters are column names: `X` is
+            the column on the 2010 geometries which contains the relevant CVAP
+            data; `Y` is the column on the 2010 geometries which contains the
+            relevant VAP data; `Z` is the column on the 2020 geometries to be
+            weighted by the ratio of the per-unit ratios in `X` and `Y`. For
+            example, if we wish to estimate Black CVAP, this triple would be
+            `(NHBCVAP19, BVAP19, BVAP20)`, which takes the ratios of the `NHBCVAP19`
+            and `BVAP19` columns on the 2010 geometries, and multplies the 2020
+            geometries' respective `BVAP20` values by these ratios.
+        ceiling (float): Number representing where to cap the weighting ratio of
+            CVAP to VAP20. After this percentage ceiling is passed, the percentage
+            will be set to 1. We recommend setting this to 1.
+        zfill (float): Fill in ratio for CVAP to VAP20 when there is 0 CVAP in the
+            area. We recommend setting this parameter to `0.1`.
+        geometry10 (str, optional): The 2010 geometry on which cvap will be pulled.
+            Acceptable values are `"tract"` or `"block group"`. As tracts are
+            less susceptible to change across Census vintages, setting this parameter
+            to `"tract"` is recommended, as it is more likely that the 2020 Census
+            blocks fit neatly into the 2010 Census tracts.
+
+    Returns: 
+       `base` geometries with 2019 CVAP-weighted 2020 CVAP estimates attached.
+    """
+    if geometry10 not in {"block group", "tract"}:
+        print(f"Requested geometry \"{geometry10}\" is not allowed; loading tracts.")
+        geometry10 = "tract"
+
+    # Grab ACS and CVAP special-tab data, and make sure our triples are correct
+    cvap_geoid = "TRACT10" if geometry10 == "tract" else "BLOCKGROUP10"
+    acs_source = acs5(state, geometry10)
+    cvap_source = cvap(state, geometry10)
+
+    # Validate the columns passed, issuing user warnings when it's inadvisable
+    # to estimate CVAP given the passed columns.
+    for (cvap10, vap10, vap20) in groups:
```

Comment:
Just adding some returns so formatting is consistent across machines and doesn't get wrapped accidentally!